cmd/pfcpsim-client/main.go

```diff
@@ -0,0 +1,379 @@
+package main
+
+import (
+	"fmt"
+	p4rtc "github.com/antoninbas/p4runtime-go-client/pkg/client"
+	"github.com/antoninbas/p4runtime-go-client/pkg/util/conversion"
+	"github.com/omec-project/upf-epc/pkg/mockSmf/smf"
+	"github.com/omec-project/upf-epc/test/integration/providers"
+	p4_v1 "github.com/p4lang/p4runtime/go/p4/v1"
+	"github.com/pborman/getopt/v2"
+	"github.com/sirupsen/logrus"
+	"io"
+	"net"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+	"sync"
+	"time"
+)
+
+var (
+	log *logrus.Logger
+
+	remotePeerAddress net.IP
+	localAddress      net.IP
+	upfAddress        net.IP
+	NodeBAddress      net.IP
+	ueAddressPool     string
+
+	inputFile string
+	doOnce    sync.Once
+
+	sessionCount int
+
+	globalMockSmf *smf.MockSMF
+)
+
+const (
+	// Values for mock-up4 environment
+	defaultGNodeBAddress = "198.18.0.10"
+	defaultUeAddressPool = "17.0.0.0/24"
+
+	defaultUpfN3Address = "198.18.0.1"
+
+	defaultSliceID = 0
+
+	srcIfaceAccess = 0x1
+	srcIfaceCore   = 0x2
+
+	directionUplink   = 0x1
+	directionDownlink = 0x2
+)
+
+// GetLoggerInstance uses a singleton pattern to return a single logger instance
+// that can be used anywhere.
+func GetLoggerInstance() *logrus.Logger {
+	// setting global logging instance
+	doOnce.Do(func() {
+		log = logrus.New()
+	})
+	return log
+}
+
+func SetLogLevel(level logrus.Level) {
+	log.SetLevel(level)
```

Comment:
this can be moved directly to the `main` func - no need to create a separate function.